app/src/main/java/com/arctouch/io/outdoorsychallenge/features/searchrv/SearchRvViewModel.kt

```diff
@@ -46,6 +53,17 @@ class SearchRvViewModel(private val repository: IVehicleRepository, dispatcherMa
 
     fun onSwipeToRefresh() = pagingDataSource.invalidate()
 
+    fun getFavoritesVehicles() {
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
+                val vehicles = repository.getFavoriteVehicles()
+                withContext(Dispatchers.Main) {
+                    _favoriteVehicles.value = vehicles
+                }
+            }
+        }
```

Comment:
this guy is on Main by default, so you could do it like:
```suggestion
        viewModelScope.launch {
            val vehicles = withContext(Dispatchers.IO) { repository.getFavoriteVehicles() }
            _favoriteVehicles.postValue(vehicles)
        }
```
because as it's a suspend function, the rest of the body of the launch will wait for the execution of the withContext